### PR TITLE
[s] FUCKING GHOST FRIENDS

### LINKED ISSUE #16909: extinguisher cabinet can be interacted with by ghosts
Ghosts can open and close extinguisher cabinet's with Alt+Click when orbiting object

This occurred on Basil, on MetaStation.
Appears to be all cabinets.

Steps:
-Be a ghost
-Double click/double alt click on a cabinet so that you are now orbiting it
-Alt click on cabinet
-It opens at closes.

### PR DESCRIPTION
**Pull request by PKPenguin321**
GET THE FUCK OFF MY FIRE EXTINGUISHERS

REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE

Fixes https://github.com/tgstation/-tg-station/issues/16909

### CODE REVIEW

File: code/game/objects/structures/extinguisher.dm [hunk 1]
```diff
@@ -68,7 +68,7 @@
 	return
 
 /obj/structure/extinguisher_cabinet/AltClick(mob/living/user)
-	if(user.incapacitated() || !Adjacent(user))
+	if(user.incapacitated() || !Adjacent(user) || !ishuman(user))
```
Comment (RemieRichards) [inline code/game/objects/structures/extinguisher.dm]: ```!istype(user)``` would be nicer, yeah animals would be able to use it, but so would drones and stuff, which atm you're excluding (robots too)

**Comment by jughu** [comment]
WHY CANT WE GHOSTS HAUNT YOU IN PIECE

File: code/game/objects/structures/extinguisher.dm (hunk 1 above)
Comment (Incoming5643) [inline code/game/objects/structures/extinguisher.dm]: we have isliving(), don't we? (or if we don't, why not?)

I'm not sure if calling isliving() is any more or less expensive than !istype(). I'm going to guess that it probably is more or less the same thing.

Comment (RemieRichards) [inline code/game/objects/structures/extinguisher.dm]: ```isliving()``` ans the other ```isX()``` we've defined were changed to macros a while ago, so it's the difference between ```istype(X, /mob/living)``` and ```istype(X)``` I'm not sure which byond performs better in (if at all), explicit or implicit types.

**Comment by octareenroon91** [comment]
Please respect `config.ghost_interaction`

File: code/game/objects/structures/extinguisher.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/game/objects/structures/extinguisher.dm]: so wait which do i use

Comment (optimumtact) [inline code/game/objects/structures/extinguisher.dm]: istype(X)

**Comment by PKPenguin321** [comment]
i might have fixed it maybe i am not sure

**Comment by optimumtact** [comment]
>Please respect config.ghost_interaction
Does literally anything respect that?

**Comment by octareenroon91** [comment]
Honestly I have no clue and downstream I have ghosts complaining of lack of Boo! and chair rotation.

**Comment by Cheridan** [comment]
that config is a meme

**Pull request merged**